system-addon/content-src/components/TopSites/TopSiteForm.jsx

```diff
@@ -38,44 +39,25 @@ export class TopSiteForm extends React.PureComponent {
 
     if (this.validateForm()) {
       const site = {url: this.cleanUrl()};
+      const index = this.props.index || -1;
```

Comment:
This will incorrectly make edits to the `0` index become inserts. We should just `const {index} = this.props;`